assignments/array-methods.js

```diff
@@ -55,29 +56,128 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
-let fullName = [];
-console.log(fullName);
+
+// let fullName = [];
+// for (let i = 0; i < runners.length; i++) {
+//     fullName.push(runners[i].first_name + " " + runners[i].last_name)
+// }
+// console.log(fullName)
+
+
+let fullName2 = []
+runners.forEach(runner =>  fullName2.push(`${runner.first_name} ${runner.last_name}`))
+console.log(fullName2);
+
+
+let fullName4 = []
+runners.forEach(addName)
+
+function addName(word){
+    fullName4.push(`${word.first_name} ${word.last_name}`)
+}
+
+console.log(fullName4)
+
+
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
-console.log(allCaps); 
+
+// let allCaps = [];
+let allCaps = runners.map(allCapMode)
+
+function allCapMode (words) {
+    return words.first_name.toUpperCase()
+    // allCaps.push(`${wubba.first_name}`)
+}
+
+
+
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
-// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+// The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. 
+//Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
+
+
+let largeShirts = runners.filter(shirts).map(names)
+
+function shirts(words){
+if (words.shirt_size === "L") {
+    return true
+}
+else {
+    return false
+}
+}
+
+function names(xxx) {
+    return xxx.first_name + " - " + xxx.shirt_size
+}
+
 console.log(largeShirts);
 
+
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+
+
+
+ let ticketPriceTotal = runners.reduce(sums, 0)
+
+function sums(x, y) {
+    return x + y.donation
+}
```

Comment:
In reducers, you are better off sticking to keeping the actual callback inside of your `reduce` and in 1 unit.